File: 1575-maximum-area-of-a-piece-of-cake-after-horizontal-and-vertical-cuts/1575-maximum-area-of-a-piece-of-cake-after-horizontal-and-vertical-cuts.py
class Solution:
    def maxArea(self, h: int, w: int, horizontalCuts: List[int], verticalCuts: List[int]) -> int:
        horizontalCuts.insert(0,0)
        horizontalCuts.append(h)
        verticalCuts.insert(0,0)
        verticalCuts.append(w)
        horizontalCuts.sort()
        verticalCuts.sort()
        hmaxi = 0
        for i in range(len(horizontalCuts)-1):
            hmaxi = max(hmaxi,horizontalCuts[i+1]-horizontalCuts[i])
            # print(i,hmaxi)
        # print('-------------')
        vmaxi = 0
        for i in range(len(verticalCuts)-1):
            vmaxi = max(vmaxi,verticalCuts[i+1]-verticalCuts[i])
            # print(i,vmaxi)
        # print(vmaxi,hmaxi)
        return (vmaxi*hmaxi)%(10**9+7)

File: 1575-maximum-area-of-a-piece-of-cake-after-horizontal-and-vertical-cuts/1575-maximum-area-of-a-piece-of-cake-after-horizontal-and-vertical-cuts.java
class Solution {
    public int maxArea(int h, int w, int[] horizontalCuts, int[] verticalCuts) {
        List<Integer> hCuts = new ArrayList<>();
        for (int cut : horizontalCuts) hCuts.add(cut);
        hCuts.add(0);
        hCuts.add(h);
        Collections.sort(hCuts);

        List<Integer> vCuts = new ArrayList<>();
        for (int cut : verticalCuts) vCuts.add(cut);
        vCuts.add(0);
        vCuts.add(w);
        Collections.sort(vCuts);

        int hmaxi = 0;
        for(int i=0;i<hCuts.size()-1;i++) {
            hmaxi = Math.max(hmaxi,hCuts.get(i + 1) - hCuts.get(i));
        }

        int vmaxi = 0;
        for(int i=0;i<vCuts.size()-1;i++){
            vmaxi = Math.max(vmaxi,vCuts.get(i+1)-vCuts.get(i));
        }
        long mod = 1_000_000_007L;
        return (int)((1L*vmaxi*hmaxi)%mod);
    }
}